pkg/gardenlet/controller/shoot/shoot_control_prepare_migration.go

```diff
@@ -0,0 +1,386 @@
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerIsDown),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying internal domain DNS record",
+			Fn:           flow.TaskFn(botanist.DestroyInternalDomainDNSRecord).DoIf(dnsEnabled),
+			Dependencies: flow.NewTaskIDs(waitUntilAPIServerIsDown),
+		})
+		createETCDSnapshot = g.Add(flow.Task{
+			Name:         "Create ETCD Snapshot",
+			Fn:           flow.TaskFn(botanist.CreateETCDSnapshot).DoIf(false),
+			Dependencies: flow.NewTaskIDs(prepareControlPlaneDeployments, waitUntilAPIServerIsDown),
+		})
+		scaleETCDToZero = g.Add(flow.Task{
+			Name:         "Scale ETCD to zero",
+			Fn:           flow.TaskFn(botanist.ScaleETCDToZero).DoIf(nonTerminatingNamespace),
+			Dependencies: flow.NewTaskIDs(createETCDSnapshot),
+		})
+		cleanManagedResourcesFinalizers = g.Add(flow.Task{
+			Name:         "Clean managed resources finalizers",
+			Fn:           flow.TaskFn(botanist.CleanManagedResourcesFinalizers),
+			Dependencies: flow.NewTaskIDs(prepareControlPlaneDeployments, waitUntilResourceManagerIsDown),
+		})
+		deleteExtensionResources = g.Add(flow.Task{
+			Name:         "Deleting extension resources",
+			Fn:           flow.TaskFn(botanist.DeleteExtensionResources).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(annotateExtensionCRDsForMigration, waitUntilAPIServerIsDown),
+		})
+		waitUntilExtensionResourcesDeleted = g.Add(flow.Task{
+			Name:         "Waiting until extension resources have been deleted",
+			Fn:           botanist.WaitUntilExtensionResourcesDeleted,
+			Dependencies: flow.NewTaskIDs(deleteExtensionResources),
+		})
+		deleteNamespace = g.Add(flow.Task{
+			Name:         "Deleting shoot namespace in Seed",
+			Fn:           flow.TaskFn(botanist.DeleteNamespace).Retry(defaultInterval),
+			Dependencies: flow.NewTaskIDs(scaleETCDToZero, waitUntilExtensionResourcesDeleted, cleanManagedResourcesFinalizers, waitUntilAPIServerIsDown),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Waiting until shoot namespace in Seed has been deleted",
+			Fn:           botanist.WaitUntilSeedNamespaceDeleted,
+			Dependencies: flow.NewTaskIDs(deleteNamespace),
+		})
+
+		f = g.Compile()
+	)
+
+	if err := f.Run(flow.Opts{Logger: o.Logger, ProgressReporter: o.ReportShootProgress, ErrorCleaner: o.CleanShootTaskError}); err != nil {
+		o.Logger.Errorf("Failed to prepare Shoot %q for migration: %+v", o.Shoot.Info.Name, err)
+		return gardencorev1beta1helper.NewWrappedLastErrors(gardencorev1beta1helper.FormatLastErrDescription(err), flow.Errors(err))
+	}
+
+	o.Logger.Infof("Successfully prepared Shoot's control plane for migration %q", o.Shoot.Info.Name)
+	return nil
+}
+
+func (c *Controller) finalizeShootPrepareForMigration(shoot *gardencorev1beta1.Shoot, o *operation.Operation) (reconcile.Result, error) {
+	if len(o.Shoot.Info.Status.UID) > 0 {
+		if err := o.DeleteClusterResourceFromSeed(context.TODO()); err != nil {
```

Comment:
Why is this not the last task in the flow above?

Reply:
If the namespace is already deleted the flow exits.